src/formatDate.js

```diff
@@ -7,8 +7,54 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dayArr = date.split(fromFormat[3]);
+  let [year, month, day] = dayArr;
+  const newFormat = [];
+
+  if (fromFormat.indexOf('YYYY') === 0 || fromFormat.indexOf('YY') === 0) {
+    [year, month, day] = dayArr;
+  } else if (
+    fromFormat.indexOf('YYYY') === 1 ||
+    fromFormat.indexOf('YY') === 1
+  ) {
+    [month, year, day] = dayArr;
+  } else if (fromFormat.indexOf('MM') === 0) {
+    [month, day, year] = dayArr;
+  } else {
+    [day, month, year] = dayArr;
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'YYYY' || toFormat[i] === 'YY') {
+      if (
+        fromFormat.some((word) => word.length > 3) === true &&
```

Comment:
Checking the length of the `fromFormat` elements to determine if the year is in `YYYY` or `YY` format is inefficient. You can directly check the format using the `indexOf` method as you did earlier.